File: cpp/invalidate.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(int, char **)
{
  auto v = vector<int> {1, 2, 3};

  for(const auto &i : v)
    for(int j = 0; j <= i; ++j) {
      cout << i << ", " << j << '\n';
      v.insert(v.begin(), i + j);
    }

  return 0;
}

File: rs/src/invalidate.rs
#[cfg(feature="never")]
mod will_fail {
    fn main() {
        let mut v = vec![1, 2, 3];
        for i in &v {
            for j in 0..(i+1) {
                println!("{}, {}", i, j);
                v.insert(0, i + j);
            }
        }
    }
}
